specs/disciplina/educator/api/swagger.yaml

```diff
@@ -0,0 +1,650 @@
+      operationId: "getStudentProofs"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: "#/parameters/StudentAddr"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Proofs"
+        404:
+          description: "Student with given address not found"
+  
+  /submissions:
+    get:
+      tags:
+      - "submissions"
+      summary: "Get all submissions"
+      description: "Gets a list of all submissions done by all students. This method is inaccessible by students."
+      operationId: "getSubmissions"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            type: "array"
+            items:
+              $ref: "#/definitions/Submission"
+    post:
+      tags:
+      - "submissions"
+      summary: "Make a new submission"
+      description: "Posts a new submission with a given body. Request body should contain valid student's signature of submission contents, otherwise an error will be raised."
+      operationId: "makeSubmission"
+      produces:
+      - "application/json"
+      consumes:
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "submission"
+        description: "A submission"
+        required: true
+        schema:
+          $ref: "#/definitions/Submission"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Submission"
+        403:
+          description: "Either submission body or submission signature is invalid"
+  
+  /submissions/{submissionHash}:
+    get:  
+      tags:
+      - "submissions"
+      summary: "Get info about a submission"
+      description: "Gets a submission data by given submission hash. Returns a 404 if a student tries to get a submission which is not their own."
+      operationId: "getSubmission"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: "#/parameters/SubmissionHash"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Submission"
+        404:
+          description: "Submission with given hash not found (or a user has no rights to look it up)"
+    delete:
+      tags:
+      - "submissions"
+      summary: "Delete a submission"
+      description: "Deletes a submission from a database. Only ungraded submissions can be deleted. Students can use this method too (though this is subject to discussion)."
+      operationId: "deleteSubmission"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: "#/parameters/SubmissionHash"
+      responses:
+        200:
+          description: "successful operation"
+        404:
+          description: "Submission with given hash not found"
+  
+  /grades:
```

Comment:
Do we need such an endpoint? It is practically a database dump.

Reply:
Not really, it only lists _grades_, and there's lots of other stuff in our database. This endpoint basically provides a way for looking at the whole private chain, like a block explorer for it.